lib/manager/terraform/lockfile/util.ts

```diff
@@ -133,6 +133,9 @@ export function writeLockUpdates(
   const lines = oldLockFileContent.split('\n');
 
   const sections: string[][] = [];
+
+  // sort updates in order of appearance in the lockfile
+  updates.sort((a, b) => a.lineNumbers.block.start - b.lineNumbers.block.start);
```

Comment:
Why do we need this? Don't we need to process  in reversed order, so line numbers don't conflict ?

Reply:
We're not replacing in in place and therefore reverse order is not necessary.    
To correctly concat the results  I order them in the order of their appearance in the original lock file.